status/condition_set.go

```diff
@@ -237,3 +239,58 @@ func (c ConditionSet) findUnhealthyDependents() []Condition {
 	})
 	return conditions
 }
+
+// Unstructured is an embedded struct type that embeds the upstream type
+type Unstructured struct {
```

Comment:
I liked @ellistarn's solution of having a method with the signature `FromUnstructured(obj unsturctured.Unstructured) status.Object` (or similar) to convert the existing type into a status object. That way, you don't have to expose an external type, since you will just be returning an interface that is already well-known. You will have to create an internal type inside of the function to get what you want, but this is really just syntactic sugar so that you don't have to export another type from this package.